core/node/base_token_adjuster/src/base_token_ratio_persister.rs

```diff
@@ -112,4 +170,99 @@ impl BaseTokenRatioPersister {
 
         Ok(id)
     }
+
+    async fn send_ratio_to_l1(
+        &self,
+        api_ratio: BaseTokenAPIRatio,
+        time_in_mempool: u32,
+    ) -> anyhow::Result<()> {
+        let fn_set_token_multiplier = self
+            .chain_admin_contract
+            .function("setTokenMultiplier")
+            .context("`setTokenMultiplier` function must be present in the ChainAdmin contract")?;
+
+        let calldata = fn_set_token_multiplier
+            .encode_input(
+                &(
+                    Token::Address(self.diamond_proxy_contract_address),
+                    Token::Uint(api_ratio.numerator.get().into()),
+                    Token::Uint(api_ratio.denominator.get().into()),
+                )
+                    .into_tokens(),
+            )
+            .context("failed encoding `setTokenMultiplier` input")?;
+
+        let nonce = (*self.eth_client)
+            .as_ref()
+            .nonce_at_for_account(
+                self.token_multiplier_setter_account_address,
+                BlockNumber::Pending,
+            )
+            .await
+            .with_context(|| "failed getting transaction count")?
+            .as_u64();
+
+        let gas_price = (*self.eth_client)
+            .as_ref()
+            .get_gas_price()
+            .await
+            .with_context(|| "failed getting gas price")?
+            .as_u64()
+            * 2;
+
+        let base_fee_per_gas = self.gas_adjuster.as_ref().get_base_fee(time_in_mempool);
```

Comment:
after seeing the whole code, you may actually not want to use get_base_fee, depending on settings it may take 10+ minutes for suggested gas price to get high enough (this method is optimized so that as little gas as possible is spent, but not for transaction inclusion speed).
Tx param provider does have a method that is optimized for transaction fast inclusion:
get_blob_tx_base_fee
Please use it instead. The name is unfortunate, you can safely make "a copy" of this method with better name and add it to TxParamsProvider trait, like get_tx_base_fee_for_fast_inclusion.
Second problem is that ethereum doesn't like resending transactions without bumping prices, see how we have to bump prices on resend in eth-sender https://github.com/matter-labs/zksync-era/blob/main/core/node/eth_sender/src/eth_fees_oracle.rs#L81

Reply:
Gotcha - all makes sense, thank you. Pushed in the changes.